pkg/engine/datasource/graphql_datasource/graphql_datasource.go

```diff
@@ -771,7 +771,7 @@ func (p *Planner) resolveNestedArgumentType(fieldName []byte) (fieldTypeRef int)
 
 	for _, inputFieldDefRef := range p.visitor.Definition.InputObjectTypeDefinitions[argTypeNode.Ref].InputFieldsDefinition.Refs {
 		if bytes.Equal(p.visitor.Definition.InputValueDefinitionNameBytes(inputFieldDefRef), fieldName) {
-			return p.visitor.Definition.ResolveListOrNameType(p.visitor.Definition.InputValueDefinitions[inputFieldDefRef].Type)
+			return p.visitor.Definition.InputValueDefinitions[inputFieldDefRef].Type
```

Comment:
@jensneuse I wanted to point out this change in particular.  The old code was "looking through" the non-null type, which I think was wrong.  All tests pass with this taken out, but I don't really understand why it was in there in the first place, so I'm not 100% confident in this change.